src/go/configgenerator/filtergen/backend_auth.go

```diff
@@ -35,38 +39,84 @@ const (
 )
 
 type BackendAuthGenerator struct {
-	// audMap is the list of unique audiences in the config.
-	audMap map[string]bool
+	// UniqueAudiences is the list of unique JWT audiences in the config.
+	UniqueAudiences map[string]bool
+
+	// AudienceBySelector lists the JWT audience for each method.
+	AudienceBySelector map[string]string
+
+	IamURL                  string
+	MetadataURL             string
+	HttpRequestTimeout      time.Duration
+	DependencyErrorBehavior string
+	BackendAuthCredentials  *options.IAMCredentialsOptions
```

Comment:
Do these have to be public?

Reply:
I decided to make everything public for now because it will help with copybara.

We will import these into google3 and write new constructors for various other config formats (i.e. TD). They will need to be in a new package in google3, so these should be publicly visible to that new package.

Reply:
> I decided to make everything public for now because it will help with copybara.
Could you explain more about this? Isn't copybara just copying text?

>We will import these into google3 and write new constructors for various other config formats (i.e. TD). They will need to be in a new package in google3, so these should be publicly visible to that new package.

If we'll have new constructors, don't we want to hide these params as private to consumers can only use constructors to build objects?

Reply:
But the constructors will be in a new package, as these new config formats will be in google3, not OSS.